src/main/java/com/todoist/pojo/Person.kt

```diff
@@ -64,15 +64,31 @@ open class Person(
         }
 
         @JvmStatic
-        fun getDefaultAvatarText(fullName: String): String {
-            val escapedName = fullName.replace(ESCAPE_PATTERN, "").trim()
-            if (escapedName.isBlank()) return "?"
+        fun getDefaultAvatarText(fullName: String?, email: String?): String {
+            val escapedName = fullName?.replace(ESCAPE_PATTERN, "")?.trim().orEmpty()
+            if (escapedName.isBlank()) {
+                return getEmailInitials(email)
+            }
 
             val first = escapedName[0].uppercaseChar()
             val lastSpace = escapedName.indexOfLast { it.isWhitespace() }
             val last =
                 if (lastSpace != -1) escapedName[lastSpace + 1].uppercaseChar().toString() else ""
             return first + last
         }
+
+        private fun getEmailInitials(email: String?): String {
+            val validEmail = email ?: "?"
+            val stringBuilder = StringBuilder()
```

Comment:
There's a `buildString` helper function in Kotlin. Tbf, it literally does `StringBuilder().apply(block)` so it's totally up to you if you want to use it or not 😅 . It's the same amount of lines too.

```
return buildString {
    val emailInitials = validEmail.substringBefore("@")
        .split(".")
        .filter { it.isNotEmpty() }
        .map { emailPart -> emailPart.first().uppercase() }
    append(emailInitials)
    if (emailInitials.size > 1) {
        append(emailInitials.lastOrNull().orEmpty())
    }
}
```